### ISSUE TITLE
CI: migrating from ti-chi-bot to prow

### ISSUE THREAD

**Issue by STRRL** [chaos-mesh/chaos-mesh#3635]
## Feature Request

ti-chi-bot is a customized prow bot, which provides many different flavors to push development progress: https://github.com/ti-community-infra/tichi/blob/master/docs/README.md

We are facing several problems with ti-chi-bot:

- uncontrolled configuration: the configuration of ti-chi-bot is stored at https://github.com/ti-community-infra/configs, and none of chaos mesh maintainers/committers has permission to update the configuration. Especially when we cut the new branch to release a new version.
- the maintaining status of ti-chi-bot is confusing: on the one hand, ti-chi-bot is still under active development, and a few commits would be pushed weekly. But the documentation, RFC, and other collaboration are not updated and transparent anymore. So it's hard to figure out if the bot would work as expected in the long future.
- over-complexed review routine: plugin is the most powerful part of the ti-chi-bot/prow bot, but many of customized plugin are NOT required with Chaos Mesh. I prefer to think there is not soo heavy developing work to force us to use a restricted rule like "issue-triage", "LGT<1/2/3>" and so on. 

I think what we need are:

- setup reviewers for PR (maybe with `OWNER`)
- merge PR automatically with enough approval or LGTM
- the ability to block the PR merging process(like `/hold` when code freezing)
- cherry-pick PR to other branches

And that could be done with the native prow bot. :)

CNCF had given us several compute resources for the CI workload, and we had build a kubernetes cluster on them, for running our test cases on ARM machines. I think we could use that kubernetes cluster to run our own prow bot.

**Comment by STRRL** [comment]
What do you think about it? @chaos-mesh/maintainers @chaos-mesh/committers

**Comment by g1eny0ung** [comment]
**TL;DR**

I agreed. 🙌🏽

> the maintaining status of ti-chi-bot is confusing: on the one hand, ti-chi-bot is still under active development, and a few commits would be pushed weekly. But the documentation, RFC, and other collaboration are not updated and transparent anymore. So it's hard to figure out if the bot would work as expected in the long future.

Yep. This is opaque to us. From my perspective, **it was developed only for the purpose of serving TiDB**. In the long run, it's not for us.

> over-complexed review routine: plugin is the most powerful part of the ti-chi-bot/prow bot, but many of customized plugin are NOT required with Chaos Mesh. I prefer to think there is not soo heavy developing work to force us to use a restricted rule like "issue-triage", "LGT<1/2/3>" and so on.

Yes. Some of the features are too heavy for us, and I think we should move to learn the Kubernetes and use the OWNER mechanism (which I strongly agree with).

**Comment by g1eny0ung** [comment]
@STRRL @cwen0 BTW, there is still another thing I think it's needed to change. Currently, the teams in our org are messy. We need to reorg the active maintainers/committers and remove someone who is not lively from the existing team.

I'm here to request org access to update it. 🙂

**Comment by g1eny0ung** [comment]
> @STRRL @cwen0 BTW, there is still another thing I think it's needed to change. Currently, the teams in our org are messy. We need to reorg the active maintainers/committers and remove someone who is not lively from the existing team.
> 
> I'm here to request org access to update it. 🙂

I have discussed this with @cwen0. The `dev` team will be removed.

**Comment by Andrewmatilde** [comment]
I agreed.

**Comment by xlgao-zju** [comment]
agree
